### DISCUSSION TITLE
Way to conditionally tell if inside a modal or not?

### DISCUSSION THREAD

**Discussion by thetwopct** [wire-elements/modal#172] [Q&A]
I am reusing some blade templates and would love to change some classes based on whether the template is inside a modal or the normal page template. Is there any way to tell if inside the modal or not? i.e. $is_modal?

**Comment by PhiloNL** [comment]
Not ideal but you could do something like this:

```
<button onclick='Livewire.emit("openModal", "edit-user", {{ json_encode(["user" => $user->id, 'modal' => true]) }})'>Edit User</button>
```

So you pass the param `modal=true` to the component, this way you can pass this variable to your view to verify if the component is inside a modal.

**Reply by thetwopct** [reply]
Thanks Philo, I'm doing exactly that and I have is_modal littered throughout my app. Thought I'd put it here as maybe there was something built in I'd missed.

**Reply by PhiloNL** [reply]
@thetwopct you could also overwrite the view and add the attribute to every component (https://github.com/wire-elements/modal/blob/main/resources/views/modal.blade.php#L50)

```php
// Before
@livewire($component['name'], $component['attributes'], key($id))

// After
@livewire($component['name'], array_merge($component['attributes'], ['modal' => true]), key($id))
```
